aspnetcore/hosting/windows-service.md

```diff
@@ -0,0 +1,105 @@
+  
+  * Call `host.RunAsService` instead of `host.Run`.
+  
+  * If your code calls `UseContentRoot`, use a path to the publish location instead of `Directory.GetCurrentDirectory()` 
+  
+  [!code-csharp[](windows-service/sample/Program.cs?name=ServiceOnly&highlight=3-4,8,14)]
+
+* Publish the application to a folder.
+
+  Use [dotnet publish](https://docs.microsoft.com/dotnet/articles/core/tools/dotnet-publish) or a Visual Studio publish profile that publishes to a folder.
+
+* Test by creating and starting the service.
+
+  Open an administrator command prompt window to use the [sc.exe](https://technet.microsoft.com/library/bb490995) command-line tool:
+
+  ```console
+  sc create servicename binPath="path\to\yourapp.exe"
+  sc start servicename
+  ```
+
+  For example, if you published to *C:\Svc* and the name of your app is AspNetCoreService, the `binPath` value would be `C:\Svc\AspNetCoreService.exe`.
+
+  ![Console window create and start example](windows-service/_static/create-start.png)
+
+  When these commands finish, you can browse to same path as when you run as a console app (by default, `http://localhost:5000`)
+
+  ![Running in a service](windows-service/_static/running-in-service.png)
+
+
+## Provide a way to run outside of a service
+
+It's easier to test when you're running outside of a service, so it's customary to add code that calls `host.RunAsService` only under certain conditions.  For example, you could run as a console app if you get a `--console` command-line argument or if the debugger is attached.
+
+[!code-csharp[](windows-service/sample/Program.cs?name=ServiceOrConsole)]
+
+## Handle stopping and starting events
+
+If you want to handle `OnStarting`, `OnStarted`, and `OnStopping` events, make the following additional changes:
+
+* Create a class that derives from `WebHostService`.
+
+  [!code-csharp[](windows-service/sample/CustomWebHostService.cs?name=NoLogging)]
+
+* Create an extension method for `IWebHost` that passes your custom `WebHostService` to `ServiceBase.Run`.
+
+  [!code-csharp[](windows-service/sample/WebHostServiceExtensions.cs?name=ExtensionsClass)]
+
+* In `Program.Main` change call the new extension method instead of `host.RunAsService`.
+
+  [!code-csharp[](windows-service/sample/Program.cs?name=HandleStopStart&highlight=26)]
+
+If your custom `WebHostService` code needs to get a service from dependency injection (such as a logger), you can get it from the `Services` property of `IWebHost`.
+
+[!code-csharp[](windows-service/sample/CustomWebHostService.cs?name=Logging&highlight=7)]
+
+## Next steps
+
+The [sample application](https://github.com/aspnet/Docs/tree/master/aspnetcore/hosting/windows-service/sample) that accompanies this article is a simple MVC web app that has been modified as shown in preceding code examples.  If you publish it to *c:\svc* (as in the included publish profile), you can run it in a service by entering these commands from an administrator window:
```

Comment:
Consider moving this to the top of the article. In this article you'll publish an ASP.NET Web app and run it as a service.  

 If you publish it to *c:\svc* (as in the included publish profile), you can run it in a service by entering these commands from an administrator window:

I had to guess at the commands

o